R/utils.R

```diff
@@ -216,23 +216,45 @@ ensure_data.table <- function(data) {
 #' dat <- as_forecast(example_quantile)
 #' print(dat)
 print.forecast_binary <- function(x, ...) {
-  # Obtain forecast object information for printing
-  forecast_type <- get_forecast_type(x)
+
+  # check whether object passes validation
+  validation <- try(do.call(validate_forecast, list(data = x)), silent = TRUE)
+  if (inherits(validation, "try-error")) {
+    validation_msg <- conditionMessage(attr(validation, "condition"))
+    cat(
+      "The object currently does not pass validation.",
+      "The following error was found:\n"
+    )
```

Comment:
```suggestion
    warning(
      "The object currently does not pass validation. ",
      "The following error was found:"
    )
```

I think this warrants a warning - but if you don't like this, use `message()`:

> [cat()](https://rdrr.io/r/base/cat.html) is for when the user asks for something to be printed and [message()](https://rdrr.io/r/base/message.html) is for when the developer elects to print something.

From: https://adv-r.hadley.nz/conditions.html

Reply:
Ah I see. I would probably opt against a warning because the user never actively called a function to validate anything, they just wanted to print. And with respect to printing, everything is fine. 

The one advantage that I see from `cat()` over a `message()` is that it is very clear where it came from. In the following example, it's a bit confusing where the messages originate from: 

<img width="828" alt="image" src="https://github.com/epiforecasts/scoringutils/assets/37978797/aa553e91-33fe-4caf-b3fd-1e53884b5427">

Reply:
Sorry, but what's different if you use `cat` here? (trying locally for me it looks the same in this instance).

Reply:
In Rstudio (I think that's due to `cli`) the colour is different!
I.e. messages are in blue, text is in black
<img width="838" alt="image" src="https://github.com/epiforecasts/scoringutils/assets/37978797/4c253b8f-c75a-4406-8e27-b8bd4ece0a6a">

Reply:
I think we should either make it such that 
- both "The object currently does not pass validation. The following error was found:" and the actual error message use `message()` or
- both use `cat()`
Happy with both!

Reply:
We did also, at some point, consider adding the number of rows that are `NA` to the print method. But that probably doesn't change anything for this question.

Reply:
> Ah I see. I would probably opt against a warning because the user never actively called a function to validate anything, they just wanted to print. And with respect to printing, everything is fine.

I confused why you wouldn't want to use a warning here? They surely only want to print it as a `as_forecast` object if it actually validates as one? If it doesn't they should be worried (i.e. a warning?)